### ISSUE TITLE
Clean up tests

### ISSUE THREAD

**Issue by amotl** [Cornices/cornice#356]
Let's drag the part about test cleanups from #351 to a new issue....

```
def test_content_type_with_callable_returning_scalar(self):
    # test that using a callable for content_type works as well
    # now, the callable returns a scalar instead of a list
    app = TestApp(main({}))
    response = app.put('/service6', headers={'Content-Type': 'audio/*'},
                       status=415)
    error_description = response.json['errors'][0]['description']
```

@almet said:
after seing this twice, I wonder if we should make it a helper function that we could put in cornice/tests/support.py instead. What do you think?

Proposal:
I'm thinking the same. Wouldn't it be good to define the services required for testing all the nitty-gritty details completely at runtime? When considering this, i can imagine targeting these topics as eventual goals of the test cleanup task:
- The _/serviceX_ jungle in `validationapp.py` would be completely cleaned up, the services will be defined at runtime from the particular test case in `test_validation.py` by using a helper function from `cornice/tests/support.py`, as you suggested. When we're already at spawning the appropriate service at test execution time, let's also revamp the uri namespace to something like _/test/service/accept_ (former uris _/service2_ and _/service3_) or _/test/service/content-type_ (former _/service5_), to name just two examples to improve on "naming things".
- Just a **single decorator-based** service definition would remain inside the revamped `validationapp.py` for testing the decorator machinery against all available http verbs on a single service uri _/test/service/decorated_.

I think it is sufficient to have the decorator-based service definition tested only once while testing all the other aspects of Cornice against services created at runtime. I believe this would cleanup and streamline the testing code and make its handling a bit easier.

Does this sound good to you? As we didn't define services at runtime before and were just using the decorator-style definition in our projects, some hints about doing this would definitively help. Currently, i can't get my head around **not** using Venusian for scanning the service decorator definitions. Thanks in advance!

**Comment by leplatrem** [comment]
I am sorry @amotl for the absence of feedback :(

Cleaning cornice test is a huge effort, and it could really reduce the amount of efforts required to contribute! What you propose sounds excellent and would work IMO. 

Usually, what is recommended is to have a test suite per feature-set, and prepare stuff in setUp. In case some setup has to be shared between several suites, I like the decorator approach, propably better than inheritance/mixins...

Thanks for tackling this topic, and in case you still feel like doing it, wish you some refactoring fun ;)

**Comment by amotl** [comment]
Hi again. I guess it will be fine to close this very stale issue. If you think it is still applicable, please re-open. Cheers!
